src/Compilers/VisualBasic/Portable/Binding/Binder_Lookup.vb

```diff
@@ -1137,11 +1161,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Return
                 End If
 
-                Dim currentResult = LookupResult.GetInstance()
-                LookupForExtensionMethods(currentResult, container, name, arity, options, binder, useSiteDiagnostics)
-                MergeInternalXmlHelperValueIfNecessary(currentResult, container, name, arity, options, binder, useSiteDiagnostics)
-                result.MergeOverloadedOrPrioritized(currentResult, checkIfCurrentHasOverloads:=False)
-                currentResult.Free()
+                tempResult.Clear()
+                LookupForExtensionMethods(tempResult, container, name, arity, options, binder, useSiteDiagnostics)
+                MergeInternalXmlHelperValueIfNecessary(tempResult, container, name, arity, options, binder, useSiteDiagnostics)
+                result.MergeOverloadedOrPrioritized(tempResult, checkIfCurrentHasOverloads:=False)
```

Comment:
note: LookupForExtensionmethods could use passing in a temp-lookupresult as well.  However, we have none to pass in since we use the one here as the main result.  Walking backwards and unwinding this, there are def cycles in lookup, meanin that the savings i get here (while good) only can help for one level of depth.  

An alternative approach that couldwork would be to actually pass along  Stack of 'temporary lookup results' that all these methods can pop/push values from when tey need it.  This would effectively ensure that LookupResults never had *any* contention in ObjectPool while the compiler was doing its work as each thread would legit have it's own stack to treat as its own pool. 

I'd like to look into that.  But after this cahnge goes in as that would likely be far more invasive.